DataStructures/AATree/AATree.cs

```diff
@@ -0,0 +1,587 @@
+                rightCheck &&
+                grandchildCheck &&
+                nonLeafChildrenCheck;
+
+            return thisNodeResult && ValidateTree(node.Left) && ValidateTree(node.Right);
+        }
+
+        /// <summary>
+        /// Checks if node is a leaf, and if so if its level is 1.
+        /// </summary>
+        /// <param name="node">The node to check.</param>
+        /// <returns>true if node passes check, false otherwise.</returns>
+        private bool CheckLeafNode(AATreeNode<TKey> node)
+        {
+            if (node.Left == null && node.Right == null &&
+                node.Level != 1)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Checks if left node's level is exactly one less than node's level.
+        /// </summary>
+        /// <param name="node">The node to check.</param>
+        /// <returns>true if node passes check, false otherwise.</returns>
+        private bool CheckLeftSubtree(AATreeNode<TKey> node)
+        {
+            if (node.Left != null && node.Level - node.Left.Level != 1)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Checks if right node's level is either equal to or one less than node's level.
+        /// </summary>
+        /// <param name="node">The node to check.</param>
+        /// <returns>true if node passes check, false otherwise.</returns>
+        private bool CheckRightSubtree(AATreeNode<TKey> node)
+        {
+            if (node.Right != null &&
+                node.Level - node.Right.Level != 1 &&
+                node.Level != node.Right.Level)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Checks if right grandchild's (right node's right node) level is less than node.
+        /// </summary>
+        /// <param name="node">The node to check.</param>
+        /// <returns>true if node passes check, false otherwise.</returns>
+        private bool CheckRightGrandChild(AATreeNode<TKey> node)
+        {
+            if (node.Right != null &&
+                node.Right.Right != null &&
+                node.Right.Level < node.Right.Right.Level)
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Checks if node is not a leaf, and if so if it has two children.
+        /// </summary>
+        /// <param name="node">The node to check.</param>
+        /// <returns>true if node passes check, false otherwise.</returns>
+        private bool CheckNonLeafChildren(AATreeNode<TKey> node)
+        {
+            if (node.Level > 1 &&
```

Comment:
Codacy found an issue: [Use the '?:' operator here.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6367315)